test/test_config.rb

```diff
@@ -9,6 +9,58 @@
 class TestConfigFile < TestConfigFileBase
   parallelize_me!
 
+  def test_default_max_threads
+    max_threads = 16
+    max_threads = 5 if RUBY_ENGINE.nil? || RUBY_ENGINE == 'ruby'
+    assert_equal max_threads, Puma::Configuration.new.default_max_threads
+  end
+
+  def test_config_loads_correct_min_threads
+    conf = Puma::Configuration.new
+    assert_equal 0, conf.options.default_options[:min_threads]
+
+    ENV['MIN_THREADS'] = '7'
```

Comment:
Setting a global and running tests in parallel threads, you're gonna have a Bad Time (as you've seen on your tests).

Maybe just removing parallelize_me will fix everything here. Put these tests into a separate test class which is not paralellized.

Reply:
Doesn't look like removing `parallelize_me!` works here. Any other ideas?

Reply:
Scratch that- extrapolated into it's own class and used the with_env helper. All seems fine.